### PR TITLE
Prepare corefx for repo consolidation

### PR DESCRIPTION
**Pull request by ViktorHofer**
As CoreFx will be a sub-repo in the new runtime repository we move everything specific into the src directory which will be the new repo root except Arcade eng files, other eng files, Documentation and cross.

### CODE REVIEW

**Comment by davidsh** [comment]
>As CoreFx will be a sub-repo in the new runtime repository

What is a "sub-repo"? Is it related to this architecture: https://stackoverflow.com/questions/35043733/how-do-i-create-nested-repositories-in-github

**Comment by ViktorHofer** [comment]
With sub-repo i mean that our current src directory will be under dotnet/runtime/src/libraries/.

File: eng/Tools.props [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
   <!-- We need to import this props file which contains PackageReferences to analyzers so that
        analyzer build assets are imported via the Tools generated props and targets -->
-  <Import Project="Analyzers.props" Condition="'$(DotNetBuildFromSource)' != 'true'" />
+  <Import Project="$(RepositoryEngineeringDir)analyzers.props" Condition="'$(DotNetBuildFromSource)' != 'true'" />
```
Comment (safern) [inline eng/Tools.props]: The file name is with upper-case. That's why you're seeing the build error.

File: src/pkg/dir.traversal.targets [hunk 2]
```diff
@@ -1,5 +1,5 @@
 <Project>
-  <Import Project="$(RepositoryEngineeringDir)dir.traversal.targets" />
+  <Import Project="..\dir.traversal.targets" />
```
Comment (safern) [inline src/pkg/dir.traversal.targets]: Why not use the `LibrariesProjectRoot` property?

File: src/publishTestAssets.proj [hunk 3]
```diff
@@ -1,9 +1,9 @@
 <Project DefaultTargets="PublishTestAssets">
```
Comment (safern) [inline src/publishTestAssets.proj]: Why are this file an others like `sentohelix.proj` moved out of the `eng` dir?

Comment (ViktorHofer) [reply src/publishTestAssets.proj]: Those two project files either implicitly or explicitly import the Directory.Build.props/targets from corefx. Means those are very repo specific so it's just natural to move them into src. I left the other files in eng as those can be shared over-time with the other parts easily.

File: src/pkg/dir.traversal.targets (hunk 2 above)
Comment (ViktorHofer) [reply src/pkg/dir.traversal.targets]: Relative paths are usually preferred.

File: eng/Tools.props (hunk 1 above)
Comment (ericstj) [inline eng/Tools.props]: casing

File: eng/restore/repoRestore.targets [hunk 4]
```diff
@@ -4,7 +4,7 @@
     <RestoreProjectStyle Condition="'$(MSBuildProjectExtension)' != '.depproj'">None</RestoreProjectStyle>
   </PropertyGroup>
 
-  <Import Project="$(RepositoryEngineeringDir)CodeAnalysis.targets" />
+  <Import Project="$(RepositoryEngineeringDir)codeAnalysis.targets" />
```
Comment (ericstj) [inline eng/restore/repoRestore.targets]: casing?

Comment (ViktorHofer) [reply eng/restore/repoRestore.targets]: My strategy is to use upper-case for Arcade API inclusions and camelCase for everything else.

File: src/Directory.Build.props [hunk 5]
```diff
@@ -1,5 +1,356 @@
 <Project>
-  <Import Project="..\Directory.Build.props" />
+  <!-- Base repo layout properties -->
```
Comment (ericstj) [inline src/Directory.Build.props]: I take it all of this here and in d.b.targets is a copy-paste of what was in the root?

Comment (ViktorHofer) [reply src/Directory.Build.props]: correct.

File: eng/restore/repoRestore.targets (hunk 4 above)
Comment (ericstj) [reply eng/restore/repoRestore.targets]: Ok, I didn't notice the file name change as part of this, wheras I did see a change for analyzers.

Comment (ViktorHofer) [reply eng/restore/repoRestore.targets]: unfortunately the git client didn't recognize the name changes and I had to do them again now...

Comment (ViktorHofer) [reply eng/restore/repoRestore.targets]: to be more precise, files which aren't common like DisableSourceControlManagement.targets and Arcade endpoints like Build.props, Tools.props, Versions.props, Signing.props and Version.Details.xml are camelCase.

Comment (ericstj) [reply eng/restore/repoRestore.targets]: `git mv -f` or `git config core.ignorecase false`  can help.

I like the convention, can we document it somewhere?

Comment (ViktorHofer) [reply eng/restore/repoRestore.targets]: I will do that in a subsequent PR when I have time. Thanks for the command :)

**Review by safern — APPROVED** [review]

**Pull request merged**